src/LibVLCSharp/Shared/Internal.cs

```diff
@@ -31,6 +30,8 @@ public abstract class Internal : IDisposable
         /// and performs the release call in native code. It will be called once when the C# instance gets disposed.</param>
         protected Internal(Func<IntPtr> create, Action<IntPtr> release)
         {
+            Core.EnsureLoaded();
```

Comment:
I'd say that this is only needed for libVLC construction. I'd put it either in this line:

```cs
        public LibVLC(params string[] options)
            : base(() => Core.EnsureLoaded(); MarshalUtils.CreateWithOptions(PatchOptions(options), Native.LibVLCNew), Native.LibVLCRelease)
```

or inside `MarshalUtils.CreateWithOptions`. What do you think ?

Reply:
Done, would probably be a good idea to document which exceptions can be thrown from `new LibVLC()` now...

Reply:
Had a discussion recently about that with a friend. Throwing from a constructor might lead to a leak if already-allocated resources are not properly freed... we'll need to be very careful about this...

Reply:
If the `LibVLC` ctor throws, a leak is probably your last problem :) but, yeah, you're right.